packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -591,7 +592,20 @@ export const Table = <
       filterTypes,
       selectSubRows,
       data,
-      getSubRows,
+      getSubRows: (originalRow: T, relativeIndex: number): T[] => {
+        if (subComponent && !originalRow.hasParent) {
+          const newSubRow = {
+            ...originalRow,
+            index: relativeIndex,
+            hasParent: true,
+            isSubComponent: true,
```

Comment:
I don't think we can add random properties to the user's row, because these could conflict with the user's data.

Reply:
I see, you're right. Users might have data with the same properties. How should I make the loop to stop in this case?

Reply:
To what I understand, getSubRows will get called when until there are no subRows left, which means it reaches the last "leaf" of the parent row. This implementation instead creates a new sub-row using the original row's data so technically it becomes its own sub-row, then gets called again as a row ready to become its own sub-row again. I added `hasParent` as a flag to mark that we already create a sub-row to be used as our sub-component. In order words, as soon as one child is created, the loop must break. 

When rendering the rows in `getPreparedRow`, when expanding row1, the expanded content is actually itself and the expanded content. The `isSubComponent` flag is there to mark that this additional row is actually a sub-component disguised as a sub-row so there is no need to render this row and just render the sub-component.
![image](https://github.com/user-attachments/assets/6f747e8c-755d-4676-a915-4633999727ef)

Reply:
I have changed the approach so these are no longer needed.